doc/number.md

```diff
@@ -0,0 +1,53 @@
+# Number
+
+A Javascript number value.
+## Methods
+
+<b>Constructors</b>
+</br>
+Create a new Javascript Number from an int, float or double.
+</br>
+<i>Parameters</i>: [Env](),  ```double```
```

Comment:
I think we agreed formatting would be similar to the existing Node.js docs which are along these lines in terms of how the parameters are defined: https://github.com/jschlight/node-addon-api/blob/ea4e5bf988ecb4a2239f93622d52c04605093a06/doc/callbackinfo.md#constructor

and https://github.com/nodejs/node-addon-api/pull/165/files